src/main/java/io/cdap/plugin/gcp/publisher/source/GoogleSubscriberConfig.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.publisher.source;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.gcp.publisher.PubSubConstants;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Extension to the PubSubSubscriberConfig class with additional fields related to record schema.
+ */
+public class GoogleSubscriberConfig extends PubSubSubscriberConfig implements Serializable {
```

Comment:
It's serializable. Is it serialized anywhere in persistent storage?

Reply:
I believe this should be because Config is used in Receiver (runs on executors). Not changed from existing code.

Reply:
I mean, if it's used in existing code that is persisted, the loading would fail because class names changed and old classes are not present.

Reply:
Ah, got it. I don't think this is persisted anywhere other than checkpoints. That should not matter since we don't support upgrade on checkpointed apps.